src/GitHub.StartPage/StartPagePackage.cs

```diff
@@ -0,0 +1,176 @@
+
+            var path = Path.Combine(request.BasePath, request.Repository.Name);
+            var uri = request.Repository.CloneUrl.ToRepositoryUrl();
+            return new CodeContainer(
+                localProperties: new CodeContainerLocalProperties(path, CodeContainerType.Folder,
+                                new CodeContainerSourceControlProperties(request.Repository.Name, path, new Guid(Guids.GitSccProviderId))),
+                remote: new RemoteCodeContainer(request.Repository.Name,
+                                                new Guid(CodeContainerProviderId),
+                                                uri,
+                                                new Uri(uri.ToString().TrimSuffix(".git")),
+                                                DateTimeOffset.UtcNow),
+                isFavorite: false,
+                lastAccessed: DateTimeOffset.UtcNow);
+        }
+
+        async Task<IGitRepositoriesExt> GetGitRepositoriesExt(IGitHubServiceProvider gitHubServiceProvider)
+        {
+            var page = await GetTeamExplorerPage(gitHubServiceProvider);
+            return page?.GetService<IGitRepositoriesExt>();
+        }
+
+        async Task<ITeamExplorerPage> GetTeamExplorerPage(IGitHubServiceProvider gitHubServiceProvider)
+        {
+            var te = gitHubServiceProvider?.GetService(typeof(ITeamExplorer)) as ITeamExplorer;
+
+            if (te != null)
+            {
+                var page = te.NavigateToPage(new Guid(TeamExplorerPageIds.Connect), null);
+
+                if (page == null)
+                {
+                    var tcs = new TaskCompletionSource<ITeamExplorerPage>();
+                    PropertyChangedEventHandler handler = null;
+
+                    handler = new PropertyChangedEventHandler((s, e) =>
+                    {
+                        if (e.PropertyName == "CurrentPage")
+                        {
+                            tcs.SetResult(te.CurrentPage);
+                            te.PropertyChanged -= handler;
+                        }
+                    });
+
+                    te.PropertyChanged += handler;
+
+                    page = await tcs.Task;
+                }
+
+                return page;
+            }
+            else
+            {
+                // TODO: Log
+                return null;
+            }
+        }
+
+        CloneRequest ShowCloneDialog(IGitHubServiceProvider gitHubServiceProvider, IGitRepositoriesExt gitRepositories, IRemoteRepositoryModel repository = null)
+        {
+            string basePath = null;
+
+            gitHubServiceProvider.AddService(this, gitRepositories);
+            var uiProvider = gitHubServiceProvider.GetService<IUIProvider>();
+            var controller = uiProvider.Configure(repository == null ? UIControllerFlow.Clone : UIControllerFlow.StartPageClone,
+                null //TODO: set the connection corresponding to the repository if the repository is not null
```

Comment:
Do we want to fix this one now? It's an edge case in case of the user being logged in to two accounts.